apps/asap-server/src/entities/event.ts

```diff
@@ -11,10 +11,10 @@ import { parseGraphQLGroup } from './group';
 
 export const parseGraphQLEvent = (item: GraphqlEvent): EventResponse => {
   const calendar = parseGraphQLCalendar(item.flatData!.calendar![0]);
-  const groups =
-    item.flatData!.calendar![0].referencingGroupsContents?.map((group) =>
-      parseGraphQLGroup(group),
-    ) || [];
+  const group =
+    item.flatData!.calendar![0].referencingGroupsContents?.map((calGroup) =>
```

Comment:
since we are only taking the first element now rather then map the items then pop the first one you could simply take the first one then call `parseGraphQLGroup` with that single element

Reply:
I started by doing that, but code was a bit more confusing. Considering that multiple references to the same calendar is the edge case, these should not have a performance impact. This way is easier to read and to test 😉

Reply:
mapping a single element in an array that is then converted to a single element is easier to read than simply invoking the function with a single argument? If you find it difficult to read - take the first element to a variable first then call the function. Your call, but i dont personally agree